code/scripts/setup/environment/superscript/superscript.sh

```diff
@@ -46,106 +49,199 @@ echo "Enter the domain name for rabbitMQ: "
+# locate the external IP section in the configuration file
 echo "Enter the server's local IP"
 read localip
 sed -i "s/10.128.0.2/$localip/" $ROOT_DIR/configurations/configurations.json
 sed -i "s/10.128.0.2/$localip/" $ENVIRONMENT_DIR/kubernetes/cluster/addons/traefik/values.yaml
 
-
+# locate the IP Allocation Range section in the script file
 echo "Enter the IP allocation range (for example: 10.32.0.0/12): "
 read IPALLOC_RANGE
 sed -i "s/172.30.0.0/16/$IPALLOC_RANGE/" $ENVIRONMENT_DIR/kubernetes/cluster/addons/weavenet/install.sh
 
 
-
-
-echo " ${blue}Installing Nano editor on the server ${reset}                 "
-
-bash $ENVIRONMENT_DIR/nano/install.sh
-
-echo " ${blue}Installing jq parser on the server ${reset}                   "
-bash $ENVIRONMENT_DIR/jq/install.sh
-
-echo " ${blue}Setting up SSL files on the server ${reset}                   "
-
-bash $ENVIRONMENT_DIR/openssl/generate.sh
-
-echo " ${blue}Installing JDK on the server ${reset}                         "
-
-bash $ENVIRONMENT_DIR/jdk/install.sh
-
-echo " ${blue}Installing Apache Maven on the server ${reset}                "
-
-bash $ENVIRONMENT_DIR/maven/install.sh
-
-echo " ${blue}Installing NodeJS on the server ${reset}                      "
-
-bash $ENVIRONMENT_DIR/nodejs/install.sh
-
-echo " ${blue}Installing NPM on the server ${reset}                         "
-
-bash $ENVIRONMENT_DIR/npm/install.sh
-
-echo " ${blue}Installing Angular on the server ${reset}                     "
-
-bash $ENVIRONMENT_DIR/angular/install.sh
-
-echo " ${blue}Installing Docker Engine CE on the server ${reset}                     "
-
-bash $ENVIRONMENT_DIR/docker/install.sh
-
-echo " ${blue}Installing Docker Compose on the server ${reset}                     "
-
-bash $ENVIRONMENT_DIR/docker/compose/install.sh
-
-echo " ${blue}Installing Docker Registry on the server ${reset}                     "
-
-bash $ENVIRONMENT_DIR/docker/registry/install.sh
-
-echo " ${blue}Installing Kubernetes on the server ${reset}                     "
-
-bash $ENVIRONMENT_DIR/kubernetes/install.sh
-
-echo " ${blue}Initializing Kubernetes Cluster on the server ${reset}                     "
-
+echo " ${blue}Installing Nano editor on the server ${reset} "
+if [ -z "$(which nano)" ] 
+then
+    echo
+    echo " ${red} nano not found, installing it ${reset} "
+    bash $ENVIRONMENT_DIR/nano/install.sh
+else 
+    echo " ${green} Package is installed! "
+    echo
+fi
+
+
+echo " ${blue}Installing jq parser on the server ${reset} "
+if [ -z "$(which jq)" ]
+then
+    echo
+    echo " ${red} jq parser not found, installing it   ${reset} "
+    bash $ENVIRONMENT_DIR/jq/install.sh
+else 
+    echo " ${green} Package is installed! "
+    echo
+fi
+
+
+echo " ${blue}Setting up SSL files on the server ${reset} "
+if [ -z "$(which openssl)" ]
+then
+    echo
+    echo " ${red} openssl not found, installing it      ${reset} "
+    bash $ENVIRONMENT_DIR/openssl/generate.sh
+else 
+    echo "${green} Package is installed!"
```

Comment:
Check my above suggestions and make the same changes at rest of the places. Don't let the user left blank understanding what package has been installed.